src/pages/resultsView/coExpression/CoExpressionVizUtils.ts

```diff
@@ -54,16 +55,23 @@ export function computePlotData(
     };
     for (const datum of mutationData) {
         if (datum.proteinChange) {
-            const targetData = (datum.entrezGeneId === xEntrezGeneId ? xMutations : yMutations);
+            const targetData = (datum.entrezGeneId === xGeneticEntityId ? xMutations : yMutations);
```

Comment:
It seems like you could avoid this loop if xGeneticEntityId is a string.

Reply:
No, because we need to check explicitly if it is a particular Entrez Gene Id. It is not about `string` or `number` here (all `datum.entrezGeneId` are of type `number`, and here we are distinguishing if the mutation is coming from the gene in X or the gene in Y).

Reply:
my point is that if xGeneticEntity is a gene, then wouldnt xGeneticEntityId be a number (the entrez gene id)? so if xGeneticEntityId is a string, then you know it is a geneset id and it seems like this whole loop is irrelevant, or redundant.

Reply:
Here you know that `xGeneticEntity` is a gene (for genesets, `mutationData.length` is 0), and what you are trying to do here is splitting which mutations correspond to the `xGeneticEntity` and which mutations correspond to the genetic entities from `y`. This is important when `xGeneticEntity` is a gene and the `yProfile` is not GSVA (so comparing against genes). Of course, if we are comparing a gene `xGeneticEntity` against GSVA, `datum.entrezGeneId === xGeneticEntityId` will always be `true` so all mutations will go to `xMutations`, and a similar thing will also happen if we are comparing a gene set `xGeneticEntity` against a "genetic" profile like mRNA (`datum.entrezGeneId === xGeneticEntityId` will always be `false` so all mutations will go to `yMutations`).

Reply:
By loop do you refer to `for (const datum of mutationData)`? If so, I get your point now

Reply:
yes that's what I mean - there's no point going through this loop if you already know that xGeneticEntityId is not an entrezGeneId. for a large query this could cause a significant slowdown. can you please add logic here to skip the loop in this case?